### ISSUE TITLE
sampsharp entity

### ISSUE THREAD

**Issue by PicoLSSSS** [ikkentim/SampSharp#337]
Someone who knows about sampsharp and tell me what is the difference between SampSharp.Gamemode and SampSharp.Entity.
which one should I use to create a gamemode?

**Comment by Sonic512** [comment]
```
 public class Program
    {
        private static void Main()
        {
            new GameModeBuilder()
                .Use<GameMode>()
                .UseLogLevel(CoreLogLevel.Info)
                .UseEncoding("C:/sa-mp/serwer/env/codepages/cp1250.txt")
                .Run();
        }
    }
```
here is example for u

**Comment by PicoLSSSSS** [comment]
> ```
>  public class Program
>     {
>         private static void Main()
>         {
>             new GameModeBuilder()
>                 .Use<GameMode>()
>                 .UseLogLevel(CoreLogLevel.Info)
>                 .UseEncoding("C:/sa-mp/serwer/env/codepages/cp1250.txt")
>                 .Run();
>         }
>     }
> ```
> 
> here is example for u

That was not my problem, my problem is what is the difference between SampSharp Gamemode and Entity and with which start of my project.

**Comment by Banditul** [comment]
Entity version follows the ECS design (just like Unity ECS) and atm there is no docs beside the grandlarc gamemode which serve as an example. 
The Gamemode is the "default" one which is the normal way of doing OOP in C#
As for what to chose, chose what you are more comfortable with. I would go with Gamemode as for now just because of the lack of docs on Entity

**Comment by PicoLSSSSS** [comment]
I asked the question because I read something about ECS being faster and I'm not sure if SampSharp.Gamemode works the same.

**Comment by PicoLSSSSS** [comment]
> Entity version follows the ECS design (just like Unity ECS) and atm there is no docs beside the grandlarc gamemode which serve as an example.
> The Gamemode is the "default" one which is the normal way of doing OOP in C#
> As for what to chose, chose what you are more comfortable with. I would go with Gamemode as for now just because of the lack of docs on Entity

I asked the question because I read something about ECS being faster and I'm not sure if SampSharp.Gamemode works the same.

**Comment by ikkentim** [comment]
It doesn't really perform differently, it's just totally different code styles. Both are fine to use, SampSharp.GameMode has been around longer and therefore has been used and tested more. SampSharp.Entities is new and I'm still working on working on good documentation for it.

**Comment by PicoLSSSSS** [comment]
> It doesn't really perform differently, it's just totally different code styles. Both are fine to use, SampSharp.GameMode has been around longer and therefore has been used and tested more. SampSharp.Entities is new and I'm still working on working on good documentation for it.

I read in the Unity forum that ESC is faster, does that apply to SampSharp.Entity?

**Comment by ikkentim** [comment]
No. Though they are similar systems, SampSharp's ECS is not related to Unity's ECS. If there is a difference in performance it is very minimal.

**Comment by PicoLSSSSS** [comment]
> No. Aunque son sistemas similares, el ECS de SampSharp no está relacionado con el ECS de Unity. Si hay una diferencia en el rendimiento, es muy mínima.

Thanks, that solves my question, I think I will use Samp Sharp.Gamemode because I couldn't find documentation for Entity
